bluesky/run_engine.py

```diff
@@ -1504,6 +1576,12 @@ def _save(self, msg):
 
             Msg('save')
         """
+        read_type = self._read_type
+        # no read type means no data saved
+        if read_type is None:
+            return
+        self._read_type = None
+        # TODO: maybe bundling should be the read_type instead of bool?
```

Comment:
I think that works, and anything that reducing the number of stateful flags is a good thing! :- )